test/apps/test_hooks.c

```diff
@@ -0,0 +1,240 @@
+/**
+ * Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2023. ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <sys/mman.h>
+#include <sys/time.h>
+#include <dlfcn.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <pthread.h>
+
+#include <ucm/api/ucm.h>
+
+
+#define _QUOTE(x) #x
+#define QUOTE(x)  _QUOTE(x)
+
+#define DEFAULT_THREAD_COUNT   16
+#define DEFAULT_TIMEOUT        10
+#define DEFAULT_THREAD_TIMEOUT 1
+
+#define log(fmt, ...) printf(fmt "\n", ##__VA_ARGS__)
+
+
+static void *mmap_call(size_t size)
```

Comment:
IMO no need this wrapper - used only once

Reply:
Putting gdb breakpoints on general mmap() function is painful as calls are everywhere. Using that function helps to simply narrow the break so i'd like to keep it?

Reply:
fixed